packages/firestore/src/core/sync_engine.ts

```diff
@@ -505,10 +504,16 @@ export class SyncEngine implements RemoteSyncer, SharedClientStateSyncer {
       // connection is disabled.
       await this.remoteStore.fillWritePipeline();
     } else if (batchState === 'acknowledged' || batchState === 'rejected') {
-      // If we receive a notification of an `acknowledged` or `rejected` batch
-      // via Web Storage, we are either already secondary or another tab has
-      // taken the primary lease.
-      await this.applyPrimaryState(false);
+      if (this.isPrimary) {
+        // If we receive a notification of an `acknowledged` or `rejected` batch
+        // via Web Storage, we are either already secondary or another tab has
+        // taken the primary lease.
+        log.debug(
+          LOG_TAG,
+          'Unexpectedly received mutation batch notification when already primary. Releasing primary lease.'
```

Comment:
If we keep this logic, we should rephrase this comment since we're not _actually_ releasing the primary lease.  We're just pretending we don't have it. :-)